integrations/astra/src/haystack_integrations/document_stores/astra/document_store.py

```diff
@@ -182,7 +179,7 @@ def write_documents(
 
         def _convert_input_document(document: Union[dict, Document]):
             if isinstance(document, Document):
-                document_dict = asdict(document)
+                document_dict = document.to_dict(flatten=False)
```

Comment:
thanks to the test introduced, I discovered that we were using the generic `asdict` instead of `Document.to_dict`. The latter knows how to properly serialize `dataframe` and `blob` fields.